multisrc/overrides/madara/olaoe/src/Olaoe.kt

```diff
@@ -0,0 +1,103 @@
+package eu.kanade.tachiyomi.extension.ar.olaoe
+
+import android.app.Application
+import android.content.SharedPreferences
+import android.widget.Toast
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.extension.BuildConfig
+import eu.kanade.tachiyomi.multisrc.madara.Madara
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.SManga
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import okhttp3.FormBody
+import okhttp3.Request
+import okhttp3.Response
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class Olaoe : Madara(
+    "Olaoe",
+    "https://olaoe.cyou",
+    "ar",
+    SimpleDateFormat("MMMM dd, yyyy", Locale("ar")),
+) {
+    private val defaultBaseUrl = "https://olaoe.cyou"
+    override val baseUrl by lazy { getPrefBaseUrl() }
+    override val mangaSubString = "works"
+    private fun getPrefBaseUrl(): String = preferences.getString(BASE_URL_PREF, defaultBaseUrl)!!
+    override fun popularMangaRequest(page: Int): Request = GET("$baseUrl/page/$page/?m_orderby=views", headers)
+    override fun chapterListSelector() = "li.wp-manga-chapter:not(.premium-block)" // Filter fake chapters
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request =
+        POST(
+            "$baseUrl/wp-admin/admin-ajax.php",
+            headers,
+            FormBody.Builder()
+                .add("action", "wp-manga-search-manga")
+                .add("title", query)
+                .build(),
+        )
+    private inline fun <reified T> Response.parseAs(): T {
+        return json.decodeFromString(body.string())
+    }
+    override fun searchMangaParse(response: Response): MangasPage {
+        val dto = response.parseAs<SearchResponseDto>()
+
+        if (!dto.success) {
+            return MangasPage(emptyList(), false)
+        }
+
+        val manga = dto.data.map {
+            SManga.create().apply {
+                setUrlWithoutDomain(it.url)
+                title = it.title
+            }
+        }
+        return MangasPage(manga, false)
+    }
+
+    @Serializable
+    data class SearchResponseDto(
+        val data: List<SearchEntryDto>,
+        val success: Boolean,
+    )
+
+    @Serializable
+    data class SearchEntryDto(
+        val url: String = "",
+        val title: String = "",
+    )
```

Comment:
Why did you need to override this? Madara search seems to work fine.

Reply:
ok thats work fine thanks